### DISCUSSION TITLE
Typing Indicator

### DISCUSSION THREAD

**Discussion by spvince** [GetStream/stream-chat-android#2073] [General]
Is it possible to display typing indicator in MessageListView as in v3 of Streamchat SDK ?

**Comment by adasiewiczr** [comment]
Hey @spvince, yes it's possible.
You need to create custom `MessageListItemViewHolderFactory`, override `createViewHolder` method and return custom view when `viewType == MessageListItemViewType.TYPING_INDICATOR` and pass it to the `MessageListView` using `setMessageViewHolderFactory` method. You can find an example of creating custom view holder factory in our [docs](https://getstream.io/chat/docs/sdk/android/ui/components/message-list/#custom-message-views).

**Reply by spvince** [reply]
Thanks
